web/src/main/java/se/inera/intyg/webcert/web/web/controller/integration/IntygIntegrationController.java

```diff
@@ -331,10 +334,10 @@ private Response buildViewCertificateResponse(UriInfo uriInfo, PrepareRedirectTo
         return Response.seeOther(location).build();
     }
 
-    private WebCertUser getWebCertUser() {
+    private WebCertUser getWebCertUser(HttpSession session) {
         final var user = getWebCertUserService().getUser();
-
         if (user.getParameters() != null && !user.getParameters().getState().hasUserBeenRedirectedToEnhetsval()) {
+            getWebCertUserService().removeSessionNow(session);
```

Comment:
Nice to remove the session in backend instead of letting the error-page in frontend do it.